### PR TITLE
Update autoscaler dependency to fix CVE

### PR DESCRIPTION
**Pull request by oksanabaza**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Bumps `k8s.io/autoscaler/cluster-autoscaler@v0.0.0-20230925095857-cf8c507d2421` to `k8s.io/autoscaler/cluster-autoscaler@v0.0.0-20240214131446-5286b3f7701a` to remove a CVE.
#### Which issue(s) this PR fixes:

Fixes # High CVEs

#### Special notes for your reviewer:

Snyk scans showed that a high CVE was introduced through: 

- `k8s.io/autoscaler/cluster-autoscaler@v0.0.0-20240214131446-5286b3f7701a` and fixed in `k8s.io/autoscaler/cluster-autoscaler@v0.0.0-20240214131446-5286b3f7701a`. 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
We better fix clusterautoscaler first and then upgrade to a version that doesn't use `satori`.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/kueue
 
-go 1.21
+go 1.21.3
+
+toolchain go1.21.6
```
Comment (alculquicondor) [inline go.mod:5]: why this discrepancy?

**Comment by ChristianZaccaria** [comment]
I think perhaps the best option is to update cluster-autoscaler to version `1.28.2`. This version uses go version 1.20 which is suitable: https://github.com/kubernetes/autoscaler/blob/cluster-autoscaler-1.28.2/cluster-autoscaler/go.mod 

Currently there is no version of cluster-autoscaler that doesn't use `satori`, so I think we can replace it for `github.com/gofrs/uuid` as discussed here: https://github.com/satori/go.uuid/issues/73#issuecomment-905005880

**Comment by tenzen-y** [comment]
> I think perhaps the best option is to update cluster-autoscaler to version `1.28.2`. This version uses go version 1.20 which is suitable: https://github.com/kubernetes/autoscaler/blob/cluster-autoscaler-1.28.2/cluster-autoscaler/go.mod
> 
> Currently there is no version of cluster-autoscaler that doesn't use `satori`, so I think we can replace it for `github.com/gofrs/uuid` as discussed here: [satori/go.uuid#73 (comment)](https://github.com/satori/go.uuid/issues/73#issuecomment-905005880)

The minimum required cluster-autoscaler version is v1.29.0 since we (kueue) depend on the provisioningRequest APIs.

**Comment by alculquicondor** [comment]
Have you send a PR to cluster-autoscaler to change the dependency? We can help prioritize the reviews there too.

**Comment by oksanabaza** [comment]
> Have you send a PR to cluster-autoscaler to change the dependency? We can help prioritize the reviews there too.

not yet, let me create a PR for cluster-autoscaler to change the dependency and I will let you know here what the PR is, thanks

**Comment by tenzen-y** [comment]
> I think perhaps the PR should be made against the branch `cluster-autoscaler-release-1.29` as opposed to master as some types we use have changed such as this [CapacityAvailable const](https://github.com/kubernetes/autoscaler/commit/ed6ebbe8baacab2fe1befd468ffaa049c66b0d73#diff-9b3a3b71dc4e27704a954d322683ec93ff459ac708971ad8f746394923a6c3adL180).
> 
> This way we can point to that branch in the go.mod. However we'll probably need to update go versions to `go1.21.3` https://github.com/kubernetes/autoscaler/blob/cluster-autoscaler-1.29.0/cluster-autoscaler/go.mod.
> 
> Thanks for the help @tenzen-y @alculquicondor ++

I don't think so since I guess that the master branch will work fine. Because some updates for the provisioning request would keep having backward compatibility.

**Comment by ChristianZaccaria** [comment]
> 
> I don't think so since I guess that the master branch will work fine. Because some updates for the provisioning request would keep having backward compatibility.

I thought it would work fine too, but we would encounter an error like such when using the latest master branch of `cluster-autoscaler`
image

Which is not encountered with 1.29 release branch. On master branch the [CapacityAvailable is removed](https://github.com/kubernetes/autoscaler/commit/ed6ebbe8baacab2fe1befd468ffaa049c66b0d73#diff-9b3a3b71dc4e27704a954d322683ec93ff459ac708971ad8f746394923a6c3adL180)

**Comment by tenzen-y** [comment]
> > I don't think so since I guess that the master branch will work fine. Because some updates for the provisioning request would keep having backward compatibility.
> 
> I thought it would work fine too, but we would encounter an error like such when using the latest master branch of `cluster-autoscaler` image
> 
> Which is not encountered with 1.29 release branch. On master branch the [CapacityAvailable is removed](https://github.com/kubernetes/autoscaler/commit/ed6ebbe8baacab2fe1befd468ffaa049c66b0d73#diff-9b3a3b71dc4e27704a954d322683ec93ff459ac708971ad8f746394923a6c3adL180)

Oh, I see. This is a nice catch, but I'm still think that updates dependencies against the master branch in the cluster-autoscaler. After that, updating the cluster-autoscaler version and adapting to new conditions in the kueue side.

Fortunately, both MultiKueue and ProvisioningACC are still in the alpha stage.

**Comment by tenzen-y** [comment]
cc @mwielgus @mimowo @alculquicondor @yaroslava-serdiuk

**Comment by yaroslava-serdiuk** [comment]
Right, CapacityAvailable doesn't have a specific condition now, it will require conditions Provisioned=True and NOT BookingExpired=True.
I think we can remove CapacityAvailable condition dependency on Kueue for now, @alculquicondor WDYT?

**Comment by oksanabaza** [comment]
> Have you send a PR to cluster-autoscaler to change the dependency? We can help prioritize the reviews there too.

Here is a PR to cluster-autoscaler to change the dependency [6554](https://github.com/kubernetes/autoscaler/pull/6554), could you please review it?

**Comment by KPostOffice** [comment]
Just a heads up, I opened another PR for `cluster-autoscaler` which upgrades the `runc` dependency

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kueue
 
-go 1.21
+go 1.21.6
```
Comment (tenzen-y) [inline go.mod:3]:
```suggestion
```

I'm not sure why we need this patch version.

Comment (alculquicondor) [reply go.mod:3]: oh shoot, we shouldn't need it.

I can send a PR in a bit.

Comment (tenzen-y) [reply go.mod:3]: One you submit PR, let me know.

**Comment by alculquicondor** [comment]
The version in the description doesn't match the code changes.
Reverting

**Comment by oksanabaza** [comment]
@alculquicondor sure, will do

**Pull request merged**